enumerable.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).each do |i|
+        yield(self[i])
+      end
+    else
```

Comment:
hello, please can you help me with an example, I am not sure how this Enumerator works.

Reply:
Okay so this simply gives enumerator so if you said `return to_enum` (this would be called on self) It'd return an enumerator.

See https://apidock.com/ruby/Object/to_enum